### PR TITLE
Add SetWinEventHook function

### PR DESCRIPTION
**Pull request by zabulus**
Please review namings, types and so on.

I'll add more documentation for `WindowsEventHookType` values tomorrow.

### CODE REVIEW

File: src/User32.Desktop/User32.cs [hunk 1]
```diff
@@ -85,6 +85,32 @@ public static partial class User32
...
+        /// <param name="hwnd">Handle to the window that generates the event, or NULL if no window is associated with the event.
+        /// For example, the mouse pointer is not associated with a window.</param>
+        /// <param name="idObject">Identifies the object associated with the event. This is one of the object identifiers or a custom object ID.
+        /// <see href="https://msdn.microsoft.com/en-us/library/windows/desktop/dd373606(v=vs.85).aspx"/></param>
+        /// <param name="idChild">Identifies whether the event was triggered by an object or a child element of the object.
+        /// If this value is CHILDID_SELF, the event was triggered by the object; otherwise, this value is the child ID of the element that triggered the event.</param>
+        /// <param name="dwEventThread">Identifies the thread that generated the event, or the thread that owns the current window.</param>
+        /// <param name="dwmsEventTime">Specifies the time, in milliseconds, that the event was generated.</param>
+        [UnmanagedFunctionPointer(CallingConvention.StdCall)]
+        public delegate void WinEventProcDelegate(
```
Comment (vbfox) [inline src/User32.Desktop/User32.cs]: The name should be just `WinEventProc` as it exists under this name in MSDN ( https://msdn.microsoft.com/en-us/library/windows/desktop/dd373885(v=vs.85).aspx )

**Review by vbfox — CHANGES_REQUESTED** [review]
Except for the delegate name, the rest LGTM.

File: src/User32.Desktop/User32.cs (hunk 1 above)
Comment (zabulus) [reply src/User32.Desktop/User32.cs]: Fixed.
BTW, for `SetWindowsHookEx` name of delegate is "WindowsHookDelegate" ( https://github.com/zabulus/pinvoke/blob/7a4af64d379380445bdd9127854ca12b3c7171e9/src/User32.Desktop/User32.cs#L85 )

Comment (vbfox) [reply src/User32.Desktop/User32.cs]: Yes sadly `WindowsHookDelegate` is as it is until we move the major version or we'll break semver :(

**Review by vbfox — APPROVED** [review]

**Comment by AArnott** [comment]
Thank you for the contribution, and @vbfox for vetting it.

**Pull request merged**